### PR TITLE
OAK-10693 | Add custom filter support for incremental FFS

### PR DESCRIPTION
**Pull request by nit0906**

### CODE REVIEW

File: oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/IndexerSupport.java [hunk 1]
```diff
@@ -230,4 +233,15 @@ public Set<String> getPreferredPathElements(Set<IndexDefinition> indexDefinition
...
     }
+
+    /**
+     *
+     * @param pattern Pattern for a custom excludes regex based on which paths would be filtered out
+     * @param typeToRepositoryPath Function to convert type <T> to valid repository path of type <String>
+     * @param <T>
+     * @return filter predicate based on a matcher for a custom excludes regex based on which paths would be filtered out
+     */
+    public <T> Predicate<T> getFilterPredicateBasedOnCustomRegex(Pattern pattern, Function<T, String> typeToRepositoryPath) {
```
Comment (steffenvan) [inline oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/IndexerSupport.java:240]: Could you write a comment about what this function is supposed to do? After looking at it shortly, I'm still not 100 % sure what problem it solves and why. Would also like an example in the Javadocs.

File: oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/IndexerSupport.java [hunk 2]
```diff
@@ -29,7 +29,10 @@
 
 import org.apache.commons.io.FileUtils;
 import org.apache.felix.inventory.Format;
-import org.apache.jackrabbit.guava.common.base.Predicate;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
```
Comment (fabriziofortino) [inline oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/IndexerSupport.java]: unused import
```suggestion
import java.util.regex.Matcher;
```

File: oak-run/src/test/java/org/apache/jackrabbit/oak/index/IncrementalStoreTest.java [hunk 3]
```diff
@@ -126,20 +142,34 @@ public void tear() {
...
+        algorithm = IndexStoreUtils.compressionAlgorithm();
+        incrementalFFSTest(false);
+        System.clearProperty(OAK_INDEXER_USE_ZIP);
     }
 
     @Test
     public void testWithGzipCompression() throws Exception {
-        algorithm = Compression.GZIP;
-        incrementalFFSTest();
+        // LZ4 compression is used by default - so disbale that, fallback is gzip
```
Comment (fabriziofortino) [inline oak-run/src/test/java/org/apache/jackrabbit/oak/index/IncrementalStoreTest.java]: typo
```suggestion
        // LZ4 compression is used by default - so disable that, fallback is gzip
```

**Review by fabriziofortino — APPROVED** [review]

File: oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/IndexerSupport.java (hunk 1 above)
Comment (nit0906) [reply oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/IndexerSupport.java:240]: changed the language a bit to make it more clear.

**Pull request merged**